test/functional/cmdline_options_tester/src/SaveOutput.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2004, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * Saveoutput matches and stores desired output from test suite in a variable
+ * 
+ * Example:
+ * 			<saveoutput regex="no" type="success" saveName="moduleAddr" splitIndex="1" splitBy="!j9module ">!j9module 0x</saveoutput>
+ * 			This matches output containing "!j9module 0x" and split the line by "!j9module " then take the second substring (Indexed "1").
+ */
+class SaveOutput extends Output {
+	private String _saveName;
+	private int _splitIndex;
+	private String _splitBy;
+	
+	/**
+	 * Pass the arguments to local variables.
```

Comment:
by "graceful error handling" do you mean output an error message, or simply fail the match?

Reply:
I mean, avoid crashing the test framework or weird exceptions like indexoutofbounds when the inputs are unexpected.  Provide as helpful a message as possible in error cases for faster triage of the problem.

in the splitBy=ajila case, a fail to match is fine, but if you think about it, it is unlikely anyone using this new feature would intend to provide a splitBy that doesn't match, so it might indicate that they set up a test that is not behaving as they expect.  Do you fail the test, or quietly ignore the "no match" and keep running a test that is maybe not testing anything?

Reply:
```
	boolean match( Object o ) {
		boolean result = super.match(o);
		if (result && (getType() == TestCondition.REQUIRED || getType() == TestCondition.SUCCESS)) {
			String line = (String) o;
			if (!line.contains(_splitBy) || _splitIndex >= line.split(_splitBy).length) {
				System.out.printf("String split failure: %n" + "String: %s%n" + "splitBy: %s%n" + "splitIndex: %d%n",
						line, _splitBy, _splitIndex);
				return false;
			}
			TestSuite.putVariable(_saveName, line.split(_splitBy)[_splitIndex]);
		}
		return result;
	}
```

does this look like the proper handle?

Reply:
You should cache the result of `line.split(_splitBy)` so you are not doing it twice.

Also, I think it would be beneficial to have a different error message for `!line.contains(_splitBy)` and ` _splitIndex >= line.split(_splitBy).length`. 

Also you need to test the `_splitIndex < 0` case

Instead of `return false;` you can set result to false. Having a single return point makes it easier to debug code.